packages/datepicker-react/src/DatePicker.tsx

```diff
@@ -156,16 +158,26 @@ export function DatePicker({
                     <Label variant={variant} forceCompact={forceCompact}>
                         {label}
                     </Label>
-                    <input
-                        placeholder={placeholder}
-                        type="text"
-                        aria-invalid={!!errorLabel}
-                        className={`jkl-text-field__input`}
-                        data-testid="jkl-datepicker__input"
-                        value={dateString}
-                        onChange={onInputChange}
-                        onFocus={openDatepicker}
-                    />
+                    <div className="jkl-text-field__input-wrapper  jkl-datepicker__input">
+                        <input
+                            placeholder={placeholder}
+                            type="text"
+                            aria-invalid={!!errorLabel}
+                            className="jkl-text-field__input"
+                            data-testid="jkl-datepicker__input"
+                            value={dateString}
+                            onChange={onInputChange}
+                        />
+                        <button
+                            aria-label={calendarButtonTitle}
+                            title={calendarButtonTitle}
+                            onClick={toggleDatepicker}
+                            data-testid="jkl-datepicker__calendar-button"
+                            className="jkl-text-field__action-button"
+                        >
+                            <CalendarIcon className="jkl-text-field__action-icon" />
+                        </button>
```

Comment:
When you put the button and input next to each-other it reads weird on screen-readers:
![Screenshot 2020-02-20 at 10 55 36](https://user-images.githubusercontent.com/124469/74922540-a8969d00-53cf-11ea-823d-2e397f7c07c0.png)

It reads the aria-label for the button and the label for the input-field at the same time as if it was one thing.

Reply:
Det er vel et resultat av at det er en `<label>´-tag rundt hele komponenten. Det henger nok igjen fra da vi først laget tekstfeltet, men er neppe det beste mønsteret for en så komplisert komponent. Det har en del bakdeler for de mindre kompliserte komponentene også, så jeg kunne godt tenkt meg å gå bort fra det over hele linjen.

Jeg kan forsøke en refaktorering her og se hvordan det funker!

Reply:
VoiceOver nekter å spille på lag på Mac-en min, så hadde vært fint om du kunne teste den nye løsningen og se om det blir bedre 😊